### PR TITLE
rargs: init at 0.3.0

### PR DESCRIPTION
**Pull request by pblkt**
###### Motivation for this change
Include [rargs](https://github.com/lotabout/rargs) in NixOS

###### Things done
Added the `rargs` derivation

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26675,6 +26675,9 @@ in
 
   py-wmi-client = callPackage ../tools/networking/py-wmi-client { };
 
+  rargs = callPackage ../tools/misc/rargs { };
+
+
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:26680]: NIT: extra newline
```suggestion
```

File: pkgs/tools/misc/rargs/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    rev    = "v${version}";
+    sha256 = "188gj05rbivci1z4z29vwdwxlj2w01v5i4avwrxjnj1dd6mmlbxd";
+  };
+
+  cargoSha256 = "0qzkhx0n28f5wy4fral3adn499q3f10q71cd544s4ghqwqn4khc9";
+
+  doCheck=false;
+
+  meta = with stdenv.lib; {
+    description = "xargs + awk with pattern matching support. `ls *.bak | rargs -p '(.*)\.bak' mv {0} {1}` ";
```
Comment (jonringer) [inline pkgs/tools/misc/rargs/default.nix:19]: should be short
```suggestion
    description = "xargs + awk with pattern matching support";
```

File: pkgs/tools/misc/rargs/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner  = "lotabout";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "188gj05rbivci1z4z29vwdwxlj2w01v5i4avwrxjnj1dd6mmlbxd";
+  };
+
+  cargoSha256 = "0qzkhx0n28f5wy4fral3adn499q3f10q71cd544s4ghqwqn4khc9";
+
+  doCheck=false;
```
Comment (jonringer) [inline pkgs/tools/misc/rargs/default.nix:16]: please comment why they are disabled

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

```
https://github.com/NixOS/nixpkgs/pull/95509
1 package built:
rargs
```

File: pkgs/tools/misc/rargs/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with stdenv.lib; {
+    description = "xargs + awk with pattern matching support. `ls *.bak | rargs -p '(.*)\.bak' mv {0} {1}` ";
+    homepage    = "https://github.com/lolabout/rargs";
+    license     = with licenses; [ mit ];
+    platforms   = platforms.all;
```
Comment (zowoq) [inline pkgs/tools/misc/rargs/default.nix:22]: `buildRustpackage` sets `meta.platforms`.
```suggestion
```

**Review by jonringer — APPROVED** [review]
sorry, one more suggestion, otherwise LGTM

```
https://github.com/NixOS/nixpkgs/pull/95509
1 package built:
rargs
```

**Review by jonringer — COMMENTED** [review]
sorry, adding a maintainer, and a package addition I consider two different actions.

The commit history should be:
```
maintainers: add pblkt
rargs: init at 0.3.0
```

to do this:
```
git reset HEAD^
git add lib
git commit -m "maintainers: add pblkt"
git add pkgs
git commit -m "rargs: init at 0.3.0"
git push ... ... --force
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 95509 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary><br>- rargs</details>

**Comment by totoroot** [comment]
@pblkt I just wanted to let you know that the provided GitHub URL (https://github.com/lolabout/rargs) is incorrect. The correct GitHub user is lotabout, therefore the URL for this package should be https://github.com/lotabout/rargs

**Pull request merged**